### PR TITLE
add Alloy app

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/roadmap/issues/3524

This PR adds Alloy application into the observability bundle for it to be used as a logging agent in place of Promtail.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add `alloy` v0.2.0
```
Comment (QuentinBisson) [inline CHANGELOG.md:12]:
```suggestion
- Add `alloy` v0.3.0
```

Could you explain a bit more why?

**Comment by hervenicol** [comment]
If it's alloy for logs, maybe we should name it `alloy-logs`?
Because we have one for the rules, one for the logs, maybe one for metrics later...
so we need to quickly understand which one does what.

File: helm/observability-bundle/values.yaml [hunk 2]
```diff
@@ -45,6 +45,30 @@ userConfig:
...
+    enabled: false
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/alloy-app
+    version: 0.3.0
+    # a list of extraConfigs for the App,
+    # It can be secret or configmap
+    # https://github.com/giantswarm/rfc/tree/main/multi-layer-app-config#example
+    extraConfigs:
+      - kind: configMap
```
Comment (QuentinBisson) [inline helm/observability-bundle/values.yaml:61]: I think we do not need the psp file

File: helm/observability-bundle/values.yaml [hunk 3]
```diff
@@ -45,6 +45,30 @@ userConfig:
...
+        name: "{{ $.Values.clusterID }}-observability-bundle-psp-override"
+        namespace: "{{ $.Release.Namespace }}"
+        priority: 150
+      - kind: secret
+        name: "{{ $.Values.clusterID }}-logging-secret"
```
Comment (QuentinBisson) [inline helm/observability-bundle/values.yaml:66]: This is expected to be the same configmap than promtail?

**Comment by QuentinBisson** [comment]
I would prefer that we do not name this alloy at all but something like logging-agentso it's easier to change the implementation if we want to

File: helm/observability-bundle/values.yaml (hunk 3 above)
Comment (TheoBrigitte) [reply helm/observability-bundle/values.yaml:66]: Yes this is expected

**Comment by TheoBrigitte** [comment]
> If it's alloy for logs, maybe we should name it `alloy-logs`? Because we have one for the rules, one for the logs, maybe one for metrics later... so we need to quickly understand which one does what.

Good point, I renamed it to `alloy-logs`.


> I would prefer that we do not name this alloy at all but something like logging-agentso it's easier to change the implementation if we want to

I don't think this make it easier for future change. There will be a need to transition from the old to the new agent, if we go through a toggle as we do now we need to have 2 distinct names. Also we can still rename this later if we want to, so I would rather continue with the current naming.

**Comment by QuentinBisson** [comment]
So I'm thinking long term here where we actually provide an api for customers and i don't think they should care about the name of the tool underneath. I find a bit weird that you talk about app name being useful for us but yet you use the same config name :D 

IMO, we should strive to have our apps named:
- logging-agent (to replace promtail)
- cluster-agent for cluster level stuff like events
- MAYBE metrics-agent 

but it's not like I'll block here, we don't have an API yet

**Comment by TheoBrigitte** [comment]
Are we fine with this then ?

**Comment by QuentinBisson** [comment]
No i am definitely not. But I definitely feel alone in that so I will not block but I will not approve

**Comment by hervenicol** [comment]
What are the implications of renaming it later?
- probably references in docs and opsrecipes will have to be updated
- does upgrade break anything?
- if configmaps and secrets change name, it means logging-operator has to know about it as well, but it seems not

So, if I didn't miss anything, the only impact is on documentation.

And as we're not sure what would be the proper name, whatever we chose we may have to change it again in the near future, right?

What if we name it `logging-agent` but the app changes again?
- we need a toggle to control when/where we do the switch
- we don't have a clear interface there, and have no clue how we would manage it for now

What if we end up with one more level of abstraction, and we have a `logging-agent` that points to `alloy-logs` but it's already named `alloy-logs`? (oops, that may be not clear, but it's not in my head either :sweat_smile: )
It's not clear to me how we should phase it, especially because AFAIK there's no plan for this interface yet.

But whether it's named `alloy-logs` or `logging-agent` now, I'm fine with both.

**Comment by QuentinBisson** [comment]
I'm reacting "a chaud" si i'll digest this and try to explain my thoughts and reasoning tomorrow :)

**Comment by QuentinBisson** [comment]
> What are the implications of renaming it later?
> 
> * probably references in docs and opsrecipes will have to be updated
> * does upgrade break anything?
> * if configmaps and secrets change name, it means logging-operator has to know about it as well, but it seems not
> 
> So, if I didn't miss anything, the only impact is on documentation.

Well, not only, this means we will have to be more careful when we rename the app and be careful with bundle versionning once again.

> And as we're not sure what would be the proper name, whatever we chose we may have to change it again in the near future, right?
>

That is true, maybe this will become node-observability-agent later on, we cannot know for sure in the long long run. I think naming tools via their feature instead of their names makes it easy to switch the underlying technology as long as the protocol stays the same. This is one of the reason otel is a good standard for instance. My current thinking is we probably should not have a mimir ingress but a monitoring ingress at the installation level so we sould not have had to switched it when we moved from Prometheus from Mimir for instance) and that I why I'm getting more interested in naming things right

> What if we name it `logging-agent` but the app changes again?
> 
> * we need a toggle to control when/where we do the switch
> * we don't have a clear interface there, and have no clue how we would manage it for now
> 
> What if we end up with one more level of abstraction, and we have a `logging-agent` that points to `alloy-logs` but it's already named `alloy-logs`? (oops, that may be not clear, but it's not in my head either 😅 ) It's not clear to me how we should phase it, especially because AFAIK there's no plan for this interface yet.
> 
> But whether it's named `alloy-logs` or `logging-agent` now, I'm fine with both.

**Comment by TheoBrigitte** [comment]
This "API" and naming convention is a different topic by itself and is out of scope for this PR. I suggest we move on with this PR and take this topic into a separate issue and go through this altogether in a meeting.

**Comment by QuentinBisson** [comment]
We should probably make this an epic cc @Rotfuks

**Review by QuentinBisson — APPROVED** [review]

**Comment by Rotfuks** [comment]
> We should probably make this an epic cc @Rotfuks

I added it to the roadmap.   
Naming and Caching...right? :D I honestly have no stakes here - I don't think at this point in our implementation the name of the component makes a difference, because we also name it grafana app and mimir and loki and all :)

**Pull request merged**